rcp/org.testeditor.rcp4.views.tcltestrun.tests/src/org/testeditor/rcp4/views/tcltestrun/model/TestLogGroupBuilderTest.xtend

```diff
@@ -0,0 +1,209 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.views.tcltestrun.model
+
+import org.testeditor.dsl.common.testing.AbstractTest
+import org.junit.Before
+import org.junit.Test
+
+class TestLogGroupBuilderTest extends AbstractTest {
+
+	TestLogGroupBuilder logGroupBuilder
+
+	@Before
+	def void setup() {
+		logGroupBuilder = new TestLogGroupBuilder
+	}
+
+	@Test
+	def void testCreateSystemComponentFirst() {
+		// given
+		val logLine = "[INFO] --- xtend-maven-plugin:2.10.0:testCompile (default) @ org.testeditor.rcp4.uatests ---"
+
+		// when
+		val group = logGroupBuilder.build(#[logLine])
+
+		// then	
+		assertTrue(group.get(0).type === TestElementType.SystemGroup)
+		assertTrue(group.get(0).logLines.contains(logLine))
+	}
+
+	@Test
+	def void testCreateComponentGroupAfterSystemComponent() {
+		// given
+		val log = '''
+			[INFO] --- xtend-maven-plugin:2.10.0:testCompile (default) @ org.testeditor.rcp4.uatests ---
+			18:49:10 TRACE [WorkbenchTestable] [TE-Test: AmlTemplateTest] AbstractTestCase  [Component] TestEditorServices
+		 '''
+
+		// when
+		val group = logGroupBuilder.build(log.split("\n"))
```

Comment:
Always use `"\r?\n"` to split strings to avoid breaking this on Windows machines.
Even better: use com.google.common.io.CharSource: `CharSource.wrap(log).readLines`